clarity/src/vm/analysis/type_checker/mod.rs

```diff
@@ -192,6 +192,28 @@ impl FunctionType {
                 }
                 Err(CheckErrors::UnionTypeError(arg_types.clone(), found_type.clone()).into())
             }
+            FunctionType::ArithmeticBinaryUIntAsSecondArg => {
```

Comment:
I'm not so pleased with this naming.. suggestions? Ideally I would almost like to have simply a Binary function type where allowed types for arg1 and arg2 can be explicitly specified... maybe for a future iteration.